api/v1alpha1/shared_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ============================================================================
+// Shared Configuration Structs
+// ============================================================================
+//
+// These structs are used across multiple resources (Cluster, Templates, Children)
+// to ensure consistency in configuration shapes.
+
+// StatelessSpec defines the desired state for a scalable, stateless component
+// like MultiAdmin, MultiOrch, or MultiGateway.
+type StatelessSpec struct {
```

Comment:
We don't have any volume setup here, is that correct?

Reply:
No, because it's stateless.

Reply:
I say we leave it as it is for now for the sake of simplicity, if in the future there is a use-case for volumes we can change it.